data-prepper-test-common/src/main/java/org/opensearch/dataprepper/test/matcher/MapEquals.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.test.matcher;
+
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeMatcher;
+
+import java.util.Map;
+
+/**
+ * Custom Hamcrest assertion {@link MapEquals#isEqualWithoutTimestamp(Map)} which compares two maps
+ * by ignoring {@link MapEquals#DEFAULT_TIMESTAMP_KEY_FOR_EVENT} key from maps.
+ * @since 1.3
+ */
+public class MapEquals extends TypeSafeMatcher<Map<String, Object>> {
+    private final Map<String, Object> expectedMap;
+    private static final String DEFAULT_TIMESTAMP_KEY_FOR_EVENT = "@timestamp";
+
+    MapEquals(Map<String, Object> map) {
+        expectedMap = map;
+    }
+
+    @Override
+    protected boolean matchesSafely(final Map<String, Object> actualMap) {
+        expectedMap.remove(DEFAULT_TIMESTAMP_KEY_FOR_EVENT);
```

Comment:
Thanks! Please copy the Map first to avoid side-effects. I believe that is the last change and I can approve.

Reply:
I'm not sure if I understood it. Does it mean to store both maps after removing timestamp?

Reply:
The caller is passing in two Map objects - actual and expected. The current code is modifying both of these maps. This is a side-effect.

Here is some quick code:

```
var modifiedExpected = new HashMap<>(expectedMap);
var modifiedActual = new HashMap<>(actualMap);
modifiedExpected.remove(...);
modifiedActual.remove(...);
return modifiedExpected.equals(actualMap);
```

You could copy the `expectedMap` and remove the timestamp in the constructor. I'm not sure this will provide any specific benefit.

You definitely should not "store" the `actualMap` after this call.